qiskit_optimization/applications/graph_optimization_application.py

```diff
@@ -0,0 +1,116 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An abstract class for graph optimization application classes."""
+
+from typing import Union, Optional, Dict, List
+from abc import abstractmethod
+
+import networkx as nx
+import numpy as np
+
+from qiskit_optimization.algorithms import OptimizationResult
+from .optimization_application import OptimizationApplication
+
+
+class GraphOptimizationApplication(OptimizationApplication):
+    """
+    An abstract class for graph optimization applications.
+    """
+
+    def __init__(self, graph: Union[nx.Graph, np.ndarray, List]) -> None:
+        """
+        Args:
+            graph: A graph representing a problem. It can be specified directly as a
+            NetworkX Graph, or as an array or list if format suitable to build out a NetworkX graph.
+        """
+        # The view of the graph is stored which means the graph can not be changed.
+        self._graph = nx.Graph(graph).copy(as_view=True)
+
+    @abstractmethod
+    def draw(self, result: Optional[Union[OptimizationResult, np.ndarray]] = None,
+             pos: Optional[Dict[int, np.ndarray]] = None) -> None:
+        """An abstract method to draw the graph based on the result.
+
+        Args:
+            result: The calculated result for the problem
+            pos: The positions of nodes
+
+        Raises:
+            NotImplementedError: When this function is not implemented in sub classes.
+        """
+        pass
```

Comment:
if each sub-class had a _draw_result() it would be even clearer I think. Base class can draw the problem, but it needs the  derived class to draw the result. I just thought it might be easier if we ever added more that each sub-class could then not have to worry about drawing the problem. But its not a needed change as such since the code would work as-is. Given whatever else you are doing I'll leave it to you as to whether its something you think is worthwhile doing now and have the time etc

Reply:
Thank you for your comments. Ya, you're right. I added `_draw_result` for future maintenance 👍